templates/sync-function/attachment-module.js

```diff
@@ -0,0 +1,132 @@
+function attachmentModule(utils, buildItemPath, resolveDocConstraint, resolveItemConstraint) {
+  var attachmentReferenceValidators = { };
+
+  return {
+    validateAttachmentReference: validateAttachmentReference,
+    validateAttachments: validateAttachments
+  };
+
+  function validateAttachmentReference(validator, itemStack, docAttachments) {
```

Comment:
From this point on, the code is largely lifted and only slightly modified from `validation-module.js`